fapolicy_analyzer/ui/profiler_page.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Concurrent Technologies Corporation 2021
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import glob
+import logging
+import gi
+import os
+
+from time import sleep
+from fapolicy_analyzer.ui.store import get_system_feature
+from fapolicy_analyzer.ui.ui_page import UIAction, UIPage
+from fapolicy_analyzer.ui.ui_widget import UIConnectedWidget
+
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+
+class ProfilerPage(UIConnectedWidget, UIPage):
+    def __init__(self):
+        UIConnectedWidget.__init__(self, get_system_feature())
+
+        actions = {
+            "start": [
+                UIAction(
+                    "Start",
+                    "Start Test",
+                    "media-playback-start",
+                    {"clicked": self.on_test_activate},
+                    sensitivity_func=self.start_button_sensitivity
+                )
+            ],
+            "stop": [
+                UIAction(
+                    "Stop",
+                    "Stop Test",
+                    "media-playback-stop",
+                    {},
+                    sensitivity_func=self.stop_button_sensitivity
+                )
+            ]
+        }
+
+        UIPage.__init__(self, actions)
+
+        self._fapd_mgr = None
+        self._fapd_profiler = None
+        self.running = False
+
+    def stop_button_sensitivity(self):
+        return self.running
+
+    def start_button_sensitivity(self):
+        return not self.running
+
+    def get_text(self):
```

Comment:
Maybe rename this method to somethin like `get_entries` that indicates you are returning something more than text. `get_text` makes me think I should expect a string back. You can also use typing to indicate what you are returning 👍